lib/protobuf/nats/client.rb

```diff
@@ -13,8 +14,85 @@ def close_connection
         # no-op (I think for now), the connection to server is persistent
       end
 
+      def self.subscription_key_cache
+        @subscription_key_cache ||= {}
+      end
+
       def send_request
-        # after method execution the "response" should be in @response_data and should be a string of bytes
+        retries ||= 3
```

Comment:
Let's roll out with it as-is so we keep the same behavior as ZMQ, we can worry about future cuts after we have some experience with it.